roles/openshift_repos/tasks/main.yaml

```diff
@@ -1,13 +1,20 @@
 ---
+- set_fact:
+    deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+    openshift_deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+
+- name: Ensure a valid deployment type has been given
+  fail:
+    msg: "Please set openshift_deployment_type to one of:\n{{ known_openshift_deployment_types | join(', ') }}"
+  when: >
+    openshift_deployment_type is not defined
```

Comment:
only if `openshift_facts` role comes before `openshift_repos` which is not guaranteed as it's not a dependency. Also it could have been defined as None...

I wanted to do this check here because it's where `known_openshift_deployment_types` is defined. But this could all be moved into a shared `openshift_inventory` role as mentioned below.